lib/src/index.js

```diff
@@ -23,6 +23,10 @@ function src(glob, opt) {
   opt.readFile = opt.read;
   opt.read = undefined;
 
+  // Leverage node-glob's statCache, conservatively
+  // TODO allow users to provide this option, at their own risk?
+  opt.statCache = Object.create(null);
```

Comment:
Agree about being wary of node-glob, I might get back to that later.

But the `statCache` implementation seems robust to me, in the sense that if we hit a match within it, those stats are reliable -- which is all I assume here.

It shouldn't increase risk of race conditions since we start with an empty object, and actually maybe decrease slightly because we keep stats consistent with what node-glob got while assembling matches.

But fair enough, let's forget about it for now.